feat(event): #3 - added base structures for CoreEvent and other essential data structures

abtinokhovat opened the pull request.

gohossein reviewed (commented):
Please change `scope` in Pull-Request title.
```
feat(core): added...
```
or
```
feat(event): added..
```

gohossein commented on .gitignore:
```diff
@@ -20,4 +20,6 @@
\ No newline at end of file
+.idea
+
+go.sum
```
I'm not sure we should include `go.sum` in `.gitignore` file.

abtinokhovat replied:
isn't this file generated in each dev environment? like package_lock.json for npm package manager?

gohossein replied:
No, this isn't lock file.
https://github.com/golang/go/wiki/Modules#releasing-modules-all-versions

abtinokhovat replied:
Done,  thanks for the documentation.
I deleted the invalid commit.

gohossein reviewed (commented):
Great start!

abtinokhovat commented:
> Please change `scope` in Pull-Request title.
> 
> ```
> feat(core): added...
> ```
> 
> or
> 
> ```
> feat(event): added..
> ```

should I add the issue number in the description?

gohossein commented:
> > Please change `scope` in Pull-Request title.
> > ```
> > feat(core): added...
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > or
> > ```
> > feat(event): added..
> > ```
> 
> should I add the issue number in the description?

Good question, I guess there is no way to directly link issue to pull-request, so I guess adding the name and link of issue in pull request description is good idea.
Actually you have access to organization and would be great if you can create a pull-request template to use for all PRs.

abtinokhovat commented:
> > > Please change `scope` in Pull-Request title.
> > > ```
> > > feat(core): added...
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > or
> > > ```
> > > feat(event): added..
> > > ```
> > 
> > 
> > should I add the issue number in the description?
> 
> Good question, I guess there is no way to directly link issue to pull-request, so I guess adding the name and link of issue in pull request description is good idea. Actually you have access to organization and would be great if you can create a pull-request template to use for all PRs.

adding issue number in PR description or link will add a hyperlink to the issue, in more simple terms we can use `#` to tag issues or PRs

gohossein commented:
Please squash your commits

gohossein commented:
Are you sure adding issue number to PR title (#3 here)  will link it to the issue?

abtinokhovat commented:
> Are you sure adding issue number to PR title (#3 here) will link it to the issue?

Yep, if you are watching this on github it will also shows you the issue when you hover on it

![CleanShot 1402-09-02 at 23 39 09](https://github.com/ormushq/ormus/assets/76223666/25a16f49-19c8-4c21-97d9-72268c77df91)

gohossein reviewed (approved):
Great
